code/go/0chain.net/smartcontract/storagesc/collect_reward.go

```diff
@@ -19,54 +19,85 @@ func (ssc *StorageSmartContract) collectReward(
 	input []byte,
 	balances cstate.StateContextI,
 ) (string, error) {
-
 	var prr stakepool.CollectRewardRequest
 	if err := prr.Decode(input); err != nil {
-		return "", common.NewErrorf("pay_reward_failed",
+		return "", common.NewErrorf("collect_reward_failed",
 			"can't decode request: %v", err)
 	}
+	if prr.ProviderType != spenum.Blobber && prr.ProviderType != spenum.Validator {
```

Comment:
call stakepool.CollectReward perhaps?

Reply:
I don't follow. There are separate smart contracts for blobbers, validators and authorises? Plus the code seems too different to merge easily.